src/lambda/commands/deleteLambda.ts

```diff
@@ -30,37 +29,38 @@ const confirm = async (message: string): Promise<boolean> => {
 }
 
 export async function deleteLambda({
+   deleteParams,
    onConfirm = async () => {
        return await confirm(localize(
            'AWS.command.deleteLambda.confirm',
            "Are you sure you want to delete lambda function '{0}'?",
-           restParams.node.configuration.FunctionName
+           deleteParams.functionName
        ))
    },
    ...restParams
 }: {
-    lambdaClient: LambdaClient
-    node: StandaloneFunctionNode, // TODO: Change to deleteParams: Lambda.Types.DeleteFunctionRequest
+    deleteParams: { functionName: string },
+    lambdaClient: Pick<LambdaClient, 'deleteFunction'>, // i.e. implements LambdaClient.deleteFunction
```

Comment:
If I understand correctly, this doesn't *implement* `deleteFunction`, it just describes the shape of an object with only that property.

Reply:
Btw, will this throw a compilation error if `deleteFunction` does not exist (for example, because it has been renamed)? If not, suggest to [guard against this](https://schneidenbach.gitbooks.io/typescript-cookbook/nameof-operator.html).

Reply:
Right, the use of `Pick` here ensures that a compilation error will occur if the object assigned to  `lambdaClient` either is missing the `deleteFunction` property or that the interface of `deleteFunction` differs from the way it's defined in `LambdaClient`. If someone were to rename the `LambdaClient.deleteFunction` to LambdaClient.delete` without looking for string replacements to notice this use of `Pick`, the compiler would be expected to warn them that LambdaClient.deleteFunction does not exist.

Reply:
The things I like about the use of `Pick` here are:

1. It informs code readers precisely what pars of `lambdaClient` are used
2. It simplifies "mocking" since only one function needs to be mocked

**Tenet**: Favor precise parameter interfaces that indicate the inputs/outputs that are used and only what's used so reader gains a basic understand what the code does without having to read the function body. 

**Bad** `doIt(params: BigGrabBag)` // when doIt only uses `firstName` and `lastName`
**Good**  `doIt(params: Pick<BigGrabBag, 'firstName' | 'lastName'>)`